src/main/java/com/couchbase/lite/replicator/PusherInternal.java

```diff
@@ -831,10 +841,11 @@ private void setPaused(boolean paused) {
 
     private void waitIfPaused(){
         synchronized (pausedObj) {
-            while (paused) {
+            while (paused && isRunning()) {
```

Comment:
Same as the comment in the ChangeTracker above, checking isRunning here could have race condition. Also chaning running status should notify pausedObj.